### PR TITLE
Reorganize documentation for MkDocs & ReadTheDocs integration

### PR DESCRIPTION
**Pull request by jakubfijalkowski**

### CODE REVIEW

File: docs/domain/ids.md [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 The domain part of the library supports a set of generic IDs:
 
-1. [`Id<T>`](../../src/Domain/LeanCode.DomainModels/Model/Id.cs),
-2. [`IId<T>`](../../src/Domain/LeanCode.DomainModels/Model/Id.cs),
-3. [`LId<T>`](../../src/Domain/LeanCode.DomainModels/Model/Id.cs),
-4. [`SId<T>`](../../src/Domain/LeanCode.DomainModels/Model/SId.cs).
+1. [`Id<T>`],
```
Comment (marcin-dardzinski) [inline docs/domain/ids.md]: Outside of the scope, but these are deprecated, docs should be adjusted.

**Review by marcin-dardzinski — APPROVED** [review]

File: docs/domain/README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # Domain
 
-1. [Value objects](../adrs/2021-02-04_value_object.md),
```
Comment (lukaszgarstecki) [inline docs/domain/README.md:3]: I believe this deserves a page as well?

File: docs/basics/mass_transit_integration.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+# MassTransit Integration
+
+[`LeanCode.CQRS.MassTransitRelay`] allows to pass raised events to [MassTransit](https://masstransit-project.com/) bus. This is the only way of consuming domain events.
```
Comment (mishioo) [inline docs/basics/mass_transit_integration.md]: Looks like shorthand reference-style links with code-formatted text doesn't work with the MkDocs, there's no link on the RedTheDocs page, only the literal [`LeanCode.CQRS.MassTransitRelay`]. I think the classic two-part reference links could do the trick:
```markdown
[`LeanCode.CQRS.MassTransitRelay`][LeanCode.CQRS.MassTransitRelay]

[LeanCode.CQRS.MassTransitRelay]: https://github.com/leancodepl/corelibrary/tree/v8.0-preview/src/CQRS/LeanCode.CQRS.MassTransitRelay
```

**Review by lukaszgarstecki — APPROVED** [review]

File: docs/basics/mass_transit_integration.md (hunk 3 above)
Comment (mishioo) [reply docs/basics/mass_transit_integration.md]: Note: the ``[`AddMassTransit`](https://masstransit.io/documentation/configuration)`` further in text works fine.

File: docs/domain/README.md (hunk 2 above)
Comment (jakubfijalkowski) [reply docs/domain/README.md:3]: It deserves, but we don't have it - the previous link was pointing at ADR which is slightly different.

File: docs/basics/mass_transit_integration.md (hunk 3 above)
Comment (jakubfijalkowski) [reply docs/basics/mass_transit_integration.md]: The problem is, I think, combining links with inline code - "just" link works. Thanks for noting!

Comment (jakubfijalkowski) [reply docs/basics/mass_transit_integration.md]: Should work now

File: docs/domain/ids.md (hunk 1 above)
Comment (jakubfijalkowski) [reply docs/domain/ids.md]: Added

**Pull request merged**